controller/work_item_events.go

```diff
@@ -0,0 +1,76 @@
+package controller
+
+import (
+	"net/http"
+
+	"github.com/fabric8-services/fabric8-wit/app"
+	"github.com/fabric8-services/fabric8-wit/application"
+	"github.com/fabric8-services/fabric8-wit/jsonapi"
+	"github.com/fabric8-services/fabric8-wit/workitem/event"
+	"github.com/goadesign/goa"
+	uuid "github.com/satori/go.uuid"
+)
+
+// WorkItemEventsController implements the work_item_events resource.
+type WorkItemEventsController struct {
+	*goa.Controller
+	db     application.DB
+	config WorkItemEventsControllerConfig
+}
+
+// WorkItemEventsControllerConfig the config interface for the WorkitemEventsController
+type WorkItemEventsControllerConfig interface {
+	//GetCacheControlWorkItemEvents() string
+	//GetCacheControlWorkItemEvent() string
+}
+
+// NewWorkItemEventsController creates a work_item_events controller.
+func NewWorkItemEventsController(service *goa.Service, db application.DB, config WorkItemEventsControllerConfig) *WorkItemEventsController {
+	return &WorkItemEventsController{
+		Controller: service.NewController("WorkItemEventsController"),
+		db:         db,
+		config:     config}
+}
+
+// List runs the list action.
+func (c *WorkItemEventsController) List(ctx *app.ListWorkItemEventsContext) error {
+	return application.Transactional(c.db, func(appl application.Application) error {
+		eventList, err := appl.Events().List(ctx, ctx.WiID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
```

Comment:
@DhritiShikhar @baijum calling `jsonapi.JSONErrorResponse` within a transaction is no longer allowed! See this issue https://github.com/openshiftio/openshift.io/issues/2689 and and #1955 from @xcoulon . I'm working on static analysis checks to find those calls.

Reply:
Fixed in 82e9588701c217f186b4db5874f09fb9c79ebb80